controllers/openstackdataplanerole_controller.go

```diff
@@ -68,9 +78,190 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, err
 	}
 
+	helper, _ := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		r.Log,
+	)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	nodes := &dataplanev1beta1.OpenStackDataPlaneNodeList{}
+
+	listOpts := []client.ListOption{
+		client.InNamespace(instance.GetNamespace()),
+	}
+	labelSelector := map[string]string{
+		"openstackdataplanerole": instance.Name,
+	}
+	if len(labelSelector) > 0 {
+		labels := client.MatchingLabels(labelSelector)
+		listOpts = append(listOpts, labels)
+	}
+	err = helper.GetClient().List(ctx, nodes, listOpts...)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if len(nodes.Items) == 0 {
+		r.Log.Info("Role: ", instance.Name, "doesn't have nodes yet, requeueing")
+		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+	}
+
+	for _, node := range nodes.Items {
+		if node.Spec.Role != instance.Name {
+			err = fmt.Errorf("node %s: node.Role does not match with node.Label", node.Name)
+			return ctrl.Result{}, err
+		}
+		if node.Status.Deployed {
+			r.Log.Info("Node: ", node.Name, "isn't deployed yet, requeueing")
+			return ctrl.Result{RequeueAfter: time.Second * 2}, nil
+		}
+	}
+
+	_, err = r.GenerateInventory(ctx, instance, nodes.Items)
+	if err != nil {
+		util.LogErrorForObject(helper, err, fmt.Sprintf("Unable to generate inventory for %s", instance.Name), instance)
+		return ctrl.Result{}, err
+	}
+
 	return ctrl.Result{}, nil
 }
 
+// GenerateInventory yields a parsed Inventory
+func (r *OpenStackDataPlaneRoleReconciler) GenerateInventory(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneRole, nodes []dataplanev1beta1.OpenStackDataPlaneNode) (string, error) {
+	var (
+		err      error
+		hostName string
+	)
+
+	inventory := ansible.MakeInventory()
+	overcloud := inventory.AddGroup("allovercloud").AddChild(ansible.MakeGroup("overcloud"))
```

Comment:
we don't want/need allovercloud/overcloud groups. I would just use the role name as the group name. There is an implicit "all" group already provided by ansible.